examples/options/include/traccc/options/details/value_array.hpp

```diff
@@ -0,0 +1,50 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// System include(s).
+#include <array>
+#include <iosfwd>
+
+namespace traccc::opts {
+
+/// A fixed number of values as one user option.
+///
+/// @note Implemented as a subclass so it is distinct from `std::array`
+///   and we can provide overloads in the same namespace.
+///
+template <typename TYPE, std::size_t SIZE>
+class value_array : public std::array<TYPE, SIZE> {};
```

Comment:
Doing so is indeed frowned upon. I would never do such a thing in a public-facing piece of code. But:
  - This code is going to stay private in this repo, even in the long term.
  - Because of how [Boost::Program_options](https://www.boost.org/doc/libs/1_84_0/doc/html/program_options.html) works, we have to use a custom class/struct to be able to define "custom streaming rules" for these variables.
  - Since this is not going to be long-term code, I really don't want to replicate much of the `std::array` interface in this custom class just to be able to use it. :thinking:

Finally, note that this PR is **not** introducing this type. I'm just renaming / rewriting the current `traccc::Reals` class.

https://github.com/acts-project/traccc/blob/main/examples/options/include/traccc/options/options.hpp

Because I was not entirely happy with either its name, or its implementation. But I did agree with its fundamental design, which @beomki-yeo came up with. :wink:

Reply:
OK, I see